### PR TITLE
executor/server/session: add extra info message sent to mysql client

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR aims to add extra info message that may be generated by `INSERT`/`REPLACE`/`UPDATE`/`LOAD` statements, for example:

```
mysql> INSERT INTO t VALUES (1,10), (3,30);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> REPLACE INTO t SELECT * from t1;
Query OK, 3 rows affected (0.00 sec)
Records: 2  Duplicates: 1  Warnings: 0

mysql> UPDATE t SET a2=2;
Query OK, 2 rows affected (0.01 sec)
Rows matched: 2  Changed: 2  Warnings: 0

mysql> LOAD DATA INFILE '/tmp/bit_test.txt' INTO TABLE bit_test;
Query OK, 2 rows affected (0.00 sec)
Records: 2  Deleted: 0  Skipped: 0  Warnings: 0
```

Notice that for each query, the last info message '`Records: ...`' was missing from previous TiDB code.

### What is changed and how it works?
In general, this PR added some counter variables to `StatementContext` and increase them during the runtime of related Executors, generate corresponding info messages when the execution is done and add the messages when TiDB send response to the client in `clientConn.writeOK() `

MySQL doesn't actually have an explanation for those messages(or I just didn't find them), so I have to check MySQL server source code and guess their meanings. I ran varieties of tests to validate the messages.

### Check List

Tests 
 - Unit test
 - Integration test

Code changes
 - No exported function/method/variable/fields change
 - Has protocol message change

Side effects
 - Slightly increased code complexity

Related changes
- No more other change

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8285)

### CODE REVIEW

File: executor/replace.go [hunk 1]
```diff
@@ -14,7 +14,9 @@
 package executor
 
 import (
+	"fmt"
```
Comment (zz-jason) [inline executor/replace.go]: add an empty line between the third-part packages.

File: executor/update.go [hunk 2]
```diff
@@ -74,19 +78,31 @@ func (e *UpdateExec) exec(schema *expression.Schema) ([]types.Datum, error) {
 			handle := row[col.Index].GetInt64()
 			oldData := row[offset:end]
 			newTableData := newData[offset:end]
+			updatable := false
 			flags := assignFlag[offset:end]
-			_, ok := e.updatedRowKeys[id][handle]
-			if ok {
+			for _, flag := range flags {
+				if flag {
+					updatable = true
```
Comment (zz-jason) [inline executor/update.go]: add a break?

**Comment by zz-jason** [comment]
@jackysp @tiancaiamao PTAL

**Comment by bb7133** [comment]
comments addressed, thanks! @zz-jason

File: executor/load_data.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ package executor
 import (
 	"bytes"
 	"fmt"
+	"github.com/pingcap/parser/mysql"
```
Comment (jackysp) [inline executor/load_data.go]: It should be moved down to the third-part imports.

File: executor/update.go [hunk 4]
```diff
@@ -14,8 +14,10 @@
 package executor
 
 import (
+	"fmt"
```
Comment (jackysp) [inline executor/update.go]: Please separate the system libs to the third-part ones.

**Comment by bb7133** [comment]
hi @jackysp , thanks for the comments, is there anything I can do to update the test cases?

**Comment by jackysp** [comment]
@bb7133 seems there is no need to fix the tests :-)
The internal test framework may ignore the extra info.
Please resolve the conflicts.

**Comment by jackysp** [comment]
PTAL @tiancaiamao @lysu

File: server/util.go [hunk 5]
```diff
@@ -343,6 +343,21 @@ func dumpTextRow(buffer []byte, columns []*ColumnInfo, row chunk.Row) ([]byte, e
 	return buffer, nil
 }
 
+func lengthEncodedIntSize(n uint64) int {
+	switch {
+	case n <= 250:
+		return 1
```
[the remark this answers is not in the document]
Comment (bb7133) [reply server/util.go]: It should be 1:
https://dev.mysql.com/doc/internals/en/integer.html#packet-Protocol::LengthEncodedInteger
You may also refer to the following codes:
https://github.com/pingcap/tidb/blob/93acd10f53c0bf6d075a518423c029b801903776/server/util.go#L62

**Comment by jackysp** [comment]
PTAL @tiancaiamao @lysu @zz-jason

**Comment by lysu** [comment]
@zz-jason  PTAL thx.

**Pull request merged**